index.js

```diff
@@ -99,7 +92,61 @@ async function pay (plugin, {
   }
 }
 
+async function pull (plugin, {
+  pointer,
+  amount,
+  streamOpts = {},
+  callback,
+  callbackOpts = {}
+}) {
+  await plugin.connect()
+  const receiveMax = amount || Infinity
+
+  let response
+  try {
+    response = await query(pointer)
+  } catch (err) {
+    const endpoint = new URL(pointer.startsWith('$')
+      ? 'https://' + pointer.substring(1)
+      : pointer)
+    const postResponse = await fetch(endpoint.href, {
+      method: 'post'
+    })
+    if (postResponse.status !== 200) {
+      throw new Error('got error response from spsp server.' +
+        ' endpoint="' + endpoint.href + '"' +
+        ' status=' + postResponse.status +
+        ' message="' + (await postResponse.text()) + '"')
+    }
+    response = await query(pointer)
+  }
+
+  if (response.contentType.indexOf('application/spsp4+json') !== -1) {
+    const ilpConn = await createConnection({
+      plugin,
+      destinationAccount: response.destinationAccount,
+      sharedSecret: response.sharedSecret,
+      ...streamOpts
+    })
+
+    const stream = await ilpConn.createStream()
+
+    try {
+      await stream.receiveTotal(receiveMax, { timeout: streamOpts.timeout })
```

Comment:
will this just time out if you can't pull the full amount? I think we should find out once the other side's sendMax is reached and then close the stream, rather than swallowing any errors.

Reply:
Would that require changing `ilp-protocol-stream`/`receiveTotal`?

Reply:
we can call `setReceiveMax` and listen for `money` events instead of using `receiveTotal`

Reply:
I'm not clear on how that'd work.
The payment could span multiple packets (due to something like `testMaximumPacketAmount`) meaning a `money` isn't necessarily representative of the sendMax

Reply:
you'd have to sum the amounts in the money events and then resolve once the total reaches the sendMax. However, with the PullError I think this is fine even if it times out.